### ISSUE TITLE
macos workflow times out

### ISSUE THREAD

**Issue by dankamongmen** [dankamongmen/notcurses#2178]
For some time now, our macOS workflow has been timing out. I'm pretty sure this is related to the new input stack. I'm able to reproduce a livelock by running `notcurses-tester` with output redirected to a file. The last errors we get are:

```
pump_control_read:994:state: 11 char: 0  48 30
pump_control_read:994:state: 11 char:    27 1b
pump_control_read:994:state:  1 char: \  92 5c
pump_control_read:994:state:  0 char:    27 1b
pump_control_read:994:state:  1 char: P  80 50
pump_control_read:994:state:  3 char: >  62 3e
pump_control_read:994:state:  4 char: | 124 7c
pump_control_read:994:state:  5 char: X  88 58
pump_control_read:994:state:  5 char: T  84 54
pump_control_rea
```

**Comment by dankamongmen** [comment]
yeah this is pretty repeatable

```

pump_control_read:994:state:  1 char: P  80 50
pump_control_read:994:state:  3 char: !  33 21
pump_control_read:994:state: 18 char: | 124 7c
pump_control_read:994:state: 19 char: 0  48 30
pump_control_read:994:state: 20 char: 0  48 30
pump_control_read:994:state: 11 char: 0  48 30
pump_control_read:994:state: 11 char: 0  48 30
pump_control_read:994:state: 11 char: 0  48 30
pump_control_read:994:state: 11 char: 0  48 30
pump_control_read:994:state: 11 char: 0  48 30
pump_control_read:994:state: 11 char: 0  48 30
pump_control_read:9
```

**Comment by dankamongmen** [comment]
hrmmm, this isn't a true lockup i'm triggering. the process eventually finishes, unlike in our workflows. but stderr does hang at this point, repeatedly. quite strange!

**Comment by dankamongmen** [comment]
```
Running tests...
11 Test project /Users/runner/work/notcurses/notcurses/build
12     Start 1: notcurses-tester
13 1/8 Test #1: notcurses-tester .................***Timeout 1500.09 sec
14     Start 2: notcurses-info
15 2/8 Test #2: notcurses-info ...................   Passed    0.16 sec
16     Start 3: ncpp_build
17 3/8 Test #3: ncpp_build .......................   Passed    0.11 sec
18     Start 4: ncpp_build_exceptions
19 4/8 Test #4: ncpp_build_exceptions ............   Passed    0.11 sec
20     Start 5: sgr-direct
21 5/8 Test #5: sgr-direct .......................***Timeout 1500.05 sec
22     Start 6: sgr-full
23 6/8 Test #6: sgr-full .........................   Passed    0.18 sec
24     Start 7: rgb
25 7/8 Test #7: rgb ..............................   Passed    0.12 sec
26     Start 8: rgbbg
27 8/8 Test #8: rgbbg ............................   Passed    0.12 sec
28 
29 75% tests passed, 2 tests failed out of 8
30 
31 Total Test time (real) = 3000.98 sec
32 
33 The following tests FAILED:
34 	  1 - notcurses-tester (Timeout)
35 	  5 - sgr-direct (Timeout)
36 Errors while running CTest
37 Output from these tests are in: 
```

**Comment by dankamongmen** [comment]
this was happily resolved by the same work that got #2174 and #2100
